c7n/filters/cost.py

```diff
@@ -0,0 +1,124 @@
+import os
+from typing import Dict
+
+from c7n.cache import InMemoryCache
+from c7n.filters import COST_ANNOTATION_KEY
+from gql import Client, gql
+from gql.transport.requests import RequestsHTTPTransport
+
+from .core import OPERATORS, Filter
+
+
+class Cost(Filter):
+    """Annotate resource monthly cost with Infracost pricing API.
+    It aims to provide an approximate cost for a generic case. For example,
+    it only grabs the on-demand price with no pre-installed software for EC2 instances.
+
+    Please use INFRACOST_API_ENDPOINT and INFRACOST_API_KEY environment vars
+    to specify the API config.
+
+    .. code-block:: yaml
+
+    policies:
+      - name: ec2-cost
+        resource: ec2
+        filters:
+          - type: cost
+            # monthly price = unit price * 730 hours
+            quantity: 730
+            op: greater-than
+            # USD
+            value: 20
+
+
+    reference: https://www.infracost.io/docs/cloud_pricing_api/overview/
+    """
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'required': ['type'],
+        'properties': {
+            'api_endpoint': {'type': 'string'},
+            'api_key': {'type': 'string'},
+            # 'currency': {'type': 'number'},
+            'quantity': {'type': 'number'},
+            'op': {'$ref': '#/definitions/filters_common/comparison_operators'},
+            'type': {'enum': ['cost']},
+            'value': {'type': 'number'},
+        },
+    }
+    schema_alias = True
+
+    def __init__(self, data, manager=None):
+        super().__init__(data, manager)
+        self.cache = InMemoryCache({})
+        self.api_endpoint = data.get(
+            "api_endpoint",
+            os.environ.get("INFRACOST_API_ENDPOINT", "https://pricing.api.infracost.io"),
+        )
+        self.api_key = data.get("api_key", os.environ.get("INFRACOST_API_KEY"))
+
+    def validate(self):
+        name = self.__class__.__name__
+        if self.api_endpoint is None:
+            raise ValueError("%s Filter requires Infracost pricing_api_endpoint" % name)
```

Comment:
generally want to use a policy validation error here

Reply:
Sorry, I don't follow the above 2 comments.
Could you please elaborate or even make the change directly?